File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/org/rustlib/vision/AprilTag.java
package org.firstinspires.ftc.teamcode.org.rustlib.vision;

import org.firstinspires.ftc.teamcode.org.rustlib.geometry.Pose3d;

import java.util.ArrayList;

public class AprilTag {
    private static ArrayList<AprilTag> tags = new ArrayList<>();
    public final Pose3d pose;
    public final int id;

    public AprilTag(Pose3d pose, int id) {
        this.pose = pose;
        this.id = id;
        tags.add(this);
    }

    public static AprilTag getTag(int id) {
        for (AprilTag tag : tags) {
            if (tag.id == id) {
                return tag;
            }
        }
        return null;
    }
}

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/common/subsystems/Apriltag.kt
package org.firstinspires.ftc.teamcode.common.subsystems

import com.arcrobotics.ftclib.command.SubsystemBase
import org.firstinspires.ftc.vision.apriltag.AprilTagDetection
import org.firstinspires.ftc.vision.apriltag.AprilTagGameDatabase
import org.firstinspires.ftc.vision.apriltag.AprilTagProcessor
import java.util.ArrayList

class Apriltag : SubsystemBase() {
    val tagLibrary = AprilTagGameDatabase.getSampleTagLibrary()

    val processor = AprilTagProcessor.Builder()
        .setDrawTagID(true)
        .setDrawTagOutline(true)
        .setDrawAxes(true)
        .setDrawCubeProjection(true)
        .setTagLibrary(tagLibrary)
        .build()

    fun getTags(): ArrayList<AprilTagDetection> {
        return processor.detections
    }
}